api/v1beta1/types_class.go

```diff
@@ -25,6 +25,10 @@ type AzureClusterClassSpec struct {
 
 	Location string `json:"location"`
 
+	// ExtendedLocation is an optional set of ExtendedLocation properties to support capz on Azure public MEC
+	// +optional
+	ExtendedLocation *ExtendedLocationSpec `json:"extendedLocation,omitempty"`
```

Comment:
In the case where `ExtendedLocation` is set, is `Location` also needed? If not, what would the user set it to since `Location`  is a required field?

Reply:
Yes, Location is needed. 
ExtendedLocation is attached to Location.